ui/oopsyraidsy/oopsyraidsy.js

```diff
@@ -964,6 +964,23 @@ class DamageTracker {
     }
   }
 
+  AddGainsEffectTriggers(type, dict) {
+    if (!dict)
+      return;
+    let keys = Object.keys(dict);
+    for (let key of keys) {
+      let id = dict[key];
+      let trigger = {
+        id: key,
+        netRegex: NetRegexes.gainsEffect({ effectId: id }),
+        mistake: function(e, data, matches) {
+          return { type: type, blame: matches.target, text: matches.effect };
```

Comment:
I think this is exactly correct.  `effect` is not filled out for `netRegex`, only for `damageRegex` / `healRegex`, so using `matches` is the right thing here.  For `netRegex`, the only thing in `event` is `line`: https://github.com/quisquous/cactbot/blob/master/ui/oopsyraidsy/oopsyraidsy.js#L647

I think we'd need to switch those special regexes over to using `netRegex` so that matches was filled out more fully.

This `EventFromFields` is a duplicate way of doing what `Regexes` does: https://github.com/quisquous/cactbot/blob/875b50b8110922c1accf8b9b7c66dce15f4436b1/ui/oopsyraidsy/oopsyraidsy.js#L806-L829.  It just predates `Regexes` (oops) and so needs to get rolled into it.